File: 605-CanPlaceFlowers/605-CanPlaceFlowers.py
# Last updated: 08/05/2026, 09:29:11
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        
        for i in range(len(flowerbed)):
            if i == 0 and len(flowerbed)==1:
                if flowerbed[i] == 0:
                    n-=1
            
            elif i == 0 and len(flowerbed)>1:
                if flowerbed[i] == 0 and flowerbed[i+1] ==0:
                    n-=1
                    flowerbed[i]=1
            elif i == len(flowerbed)-1 and len(flowerbed)>1:
                if flowerbed[i] == 0 and flowerbed[i-1] ==0:
                    n-=1
                    flowerbed[i]=1
            else:
                if flowerbed[i]==flowerbed[i+1]==flowerbed[i-1]==0:
                    n-=1
                    flowerbed[i]=1
            if n==0 or n<0:
                return True
        return False

File: 605-CanPlaceFlowers/605-CanPlaceFlowers.cpp
// Last updated: 08/05/2026, 09:29:07
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int size= flowerbed.size();
        if(n==0){
            return true;
        }
        if(size==1){
            return !flowerbed[0];
        }
        if(size==2){
            return (flowerbed[0] || flowerbed[1]) || n>1 ? false :true ;
        }


        for(int i=0; i<=size-3; i++){
            if(i==0){
                if(flowerbed[i]==0 && flowerbed[i+1]==0){
                flowerbed[i]=1;
                n-=1;
            }
            }
            if(i==size-3){
             if(flowerbed[i+1]==0 && flowerbed[i+2]==0){
                flowerbed[i+2]=1;
                n-=1;
            }   
            }
            else if(flowerbed[i]==0 && flowerbed[i+1]==0 && flowerbed[i+2]==0){
                flowerbed[i+1]=1;
                n-=1;
            }

            if(n<=0)
            return true;
        }
        
        return false;
    }
};